Buildalyzer warn and CS0518 system.attribute error

**Describe the bug**
When i run dotnet stryker on my project we go warn during the build

`[10:57:11 WRN] Buildalyzer could not find sourcefiles. This should not happen. Will fallback to filesystem scan. Please report an issue at github.`

And error

`[10:58:25 ERR] Failed to build the mutated assembly due to unrecoverable error: error CS0518: Le type prédéfini 'System.Attribute' n'est pas défini ou importé`

**Logs**
`[10:56:49 INF] Identifying project to mutate.
[10:57:11 WRN] Buildalyzer could not find sourcefiles. This should not happen. Will fallback to filesystem scan. Please report an issue at github.
[10:57:12 INF] Found project myproject.csproj to mutate.
[10:57:12 INF] Building test project myproject.Tests.csproj (1/1)
[10:58:21 INF] Total number of tests found: 157.
[10:58:21 INF] Initial testrun started.
[10:58:25 ERR] Failed to build the mutated assembly due to unrecoverable error: error CS0518: Le type prédéfini 'System.Attribute' n'est pas défini ou importé`

**Expected behavior**
No warning and build sucess

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Type of project .net framework, .net standard
 - Framework Version .net framework 4.8 / .net standard 2.0
 - Stryker Version 3.10.0

**Additional context**
To find the bug, i have used a standalone Buildalyzer to analyse my solutions (very big solution, >100 projects, multiple SNL on root directory etc...).
On stryker config, if solution isn't provide i have this message
`Stryker could not build your project as no solution file was presented. Please pass the solution path to stryker.`

Afte my fix, build logs add two additional information on skipping files

`[14:16:09 DBG] Skipping auto-generated code file: myproject\obj\Debug\.NETFramework,Version=v4.8.AssemblyAttributes.cs
[14:16:09 DBG] Skipping auto-generated code file: myproject\obj\Debug\myproject.AssemblyInfo.cs`

I made this PR to fix this https://github.com/stryker-mutator/stryker-net/pull/2683

--- a/src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/TargetFrameworkResolutionTests.cs
+++ b/src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/TargetFrameworkResolutionTests.cs
@@ -33,7 +33,7 @@ namespace Stryker.Core.UnitTest.Initialisation
             analyzerResultsMock
                 .Setup(m => m.GetEnumerator())
                 .Returns(() => _analyzerResults.GetEnumerator());
-            buildalyzerProviderMock.Setup(x => x.Provide( It.IsAny<AnalyzerManagerOptions>())).Returns(analyzerManagerMock.Object);
+            buildalyzerProviderMock.Setup(x => x.Provide(It.IsAny<string>(), It.IsAny<AnalyzerManagerOptions>())).Returns(analyzerManagerMock.Object);
 
             _projectFileReader = new ProjectFileReader(null, buildalyzerProviderMock.Object);
         }

--- a/src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs
+++ b/src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs
@@ -57,7 +57,7 @@ namespace Stryker.Core.Initialisation
                 }
                 else
                 {
-                    testProjectFileNames = new List<string> {FindTestProject(options.ProjectPath) };
+                    testProjectFileNames = new List<string> { FindTestProject(options.ProjectPath) };
                 }
 
                 var testProjects = testProjectFileNames.Select(testProjectFile => _projectFileReader.AnalyzeProject(testProjectFile, options.SolutionPath, options.TargetFramework)).ToList();
@@ -92,7 +92,7 @@ namespace Stryker.Core.Initialisation
         {
             // need to scan traverse dependencies
             // dependents contains the list of projects depending on each (non test) projects
-            var dependents = projectsUnderTest.ToDictionary(p=>p.ProjectFilePath, p => new HashSet<string>(new []{p.ProjectFilePath}));
+            var dependents = projectsUnderTest.ToDictionary(p => p.ProjectFilePath, p => new HashSet<string>(new[] { p.ProjectFilePath }));
             // register explicit dependencies
             foreach (var result in projectsUnderTest)
             {
@@ -101,7 +101,7 @@ namespace Stryker.Core.Initialisation
                     dependents[reference].Add(result.ProjectFilePath);
                 }
             }
-            
+
             // we need to dig recursively to find recursive dependencies, until none are discovered
             bool foundNewDependency;
             do
@@ -127,8 +127,8 @@ namespace Stryker.Core.Initialisation
 
         private List<IAnalyzerResult> AnalyzeSolution(StrykerOptions options)
         {
-            _logger.LogInformation("Identifying projects to mutate in {0}. This can take a while.",  options.SolutionPath);
-            var manager = _projectFileReader.AnalyzeSolution(options.SolutionPath);
+            _logger.LogInformation("Identifying projects to mutate in {0}. This can take a while.", options.SolutionPath);
+            var manager = _projectFileReader.GetAnalyzerManager(options.SolutionPath);
 
             // build all projects
             var projectsAnalyzerResults = new ConcurrentBag<IAnalyzerResult>();
@@ -171,7 +171,7 @@ namespace Stryker.Core.Initialisation
                 var projectLogName = Path.GetRelativePath(Path.GetDirectoryName(options.SolutionPath), project.ProjectFilePath);
                 var analyzerResultsForTestProjects = testProjects
                     .Where(testProject => testProject.ProjectReferences.Any(reference => dependents[project.ProjectFilePath].Contains(reference)));
-                var relatedTestProjects = analyzerResultsForTestProjects.Select(p =>p.ProjectFilePath).ToList();
+                var relatedTestProjects = analyzerResultsForTestProjects.Select(p => p.ProjectFilePath).ToList();
                 if (relatedTestProjects.Count > 0)
                 {
                     _logger.LogDebug("Matched {0} to {1} test projects:", projectLogName, relatedTestProjects.Count);
@@ -305,7 +305,7 @@ namespace Stryker.Core.Initialisation
                 stringBuilder.Append("More than one project reference matched the given project filter ")
                 .Append($"'{options.SourceProjectName}'")
                 .AppendLine(", please specify the full name of the project reference.");
-                
+
             }
 
             stringBuilder.Append(BuildReferenceChoice(projectReferences));

--- a/src/Stryker.Core/Stryker.Core/Initialisation/ProjectFileReader.cs
+++ b/src/Stryker.Core/Stryker.Core/Initialisation/ProjectFileReader.cs
@@ -16,7 +16,7 @@ namespace Stryker.Core.Initialisation
             string solutionFilePath,
             string targetFramework,
             string msBuildPath = null);
-        IAnalyzerManager AnalyzeSolution(string solutionPath);
+        IAnalyzerManager GetAnalyzerManager(string solutionPath = null);
     }
 
     /// <summary>
@@ -39,13 +39,10 @@ namespace Stryker.Core.Initialisation
             _logger = ApplicationLogging.LoggerFactory.CreateLogger<ProjectFileReader>();
         }
 
-        private IAnalyzerManager AnalyzerManager
+        public IAnalyzerManager GetAnalyzerManager(string solutionFilePath = null)
         {
-            get
-            {
-                _analyzerManager ??= _analyzerProvider.Provide(new AnalyzerManagerOptions{LogWriter = _buildalyzerLog});
-                return _analyzerManager;
-            }
+            _analyzerManager ??= _analyzerProvider.Provide(solutionFilePath, new AnalyzerManagerOptions { LogWriter = _buildalyzerLog });
+            return _analyzerManager;
         }
 
         public IAnalyzerResult AnalyzeProject(string projectFilePath,
@@ -53,9 +50,8 @@ namespace Stryker.Core.Initialisation
             string targetFramework,
             string msBuildPath = null)
         {
-
             _logger.LogDebug("Analyzing project file {0}", projectFilePath);
-            var analyzerResult = GetProjectInfo(projectFilePath, targetFramework);
+            var analyzerResult = GetProjectInfo(projectFilePath, solutionFilePath, targetFramework);
             LogAnalyzerResult(analyzerResult);
 
             if (analyzerResult.Succeeded)
@@ -67,7 +63,7 @@ namespace Stryker.Core.Initialisation
                 // buildalyzer failed to find restored packages, retry after nuget restore
                 _logger.LogDebug("Project analyzer result not successful, restoring packages");
                 _nugetRestoreProcess.RestorePackages(solutionFilePath, msBuildPath);
-                analyzerResult = GetProjectInfo(projectFilePath, targetFramework);
+                analyzerResult = GetProjectInfo(projectFilePath, solutionFilePath, targetFramework);
             }
             else
             {
@@ -78,16 +74,16 @@ namespace Stryker.Core.Initialisation
             return analyzerResult;
         }
 
-        public IAnalyzerManager AnalyzeSolution(string solutionPath) => _analyzerProvider.Provide(solutionPath, new AnalyzerManagerOptions{LogWriter = _buildalyzerLog});
-
         /// <summary>
         /// Checks if project info is already present in solution projects. If not, analyze here.
         /// </summary>
         /// <returns></returns>
-        private IAnalyzerResult GetProjectInfo(string projectFilePath,
+        private IAnalyzerResult GetProjectInfo(
+            string projectFilePath,
+            string solutionFilePath,
             string targetFramework)
         {
-            var analyzerResults = AnalyzerManager.GetProject(projectFilePath).Build();
+            var analyzerResults = GetAnalyzerManager(solutionFilePath).GetProject(projectFilePath).Build();
             return SelectAnalyzerResult(analyzerResults, targetFramework);
         }
 

--- a/src/Stryker.Core/Stryker.Core/Initialisation/ProjectFileReader.cs
+++ b/src/Stryker.Core/Stryker.Core/Initialisation/ProjectFileReader.cs
@@ -16,7 +16,7 @@ namespace Stryker.Core.Initialisation
             string solutionFilePath,
             string targetFramework,
             string msBuildPath = null);
-        IAnalyzerManager GetAnalyzerManager(string solutionPath = null);
+        IAnalyzerManager GetAnalyzerManager(string solutionFilePath = null);
     }
 
     /// <summary>

